hcc004v1_dysrhythmia_suspect.py

```diff
@@ -0,0 +1,318 @@
+
+        # TODO: Is the status field used at all for medications, or is filtering done just with start/end dates?
+        qs = Medication.objects.filter(patient__id=self.patient_id, start_date__lte=time_frame_end)
+
+        # TODO: still_active=False is not currently supported — where would the value come from?
+        still_active = True
+        if still_active:
+            return qs.filter(Q(end_date__isnull=True) | Q(end_date__gte=time_frame_end)).exists()
+        else:
+            return qs.filter(Q(end_date__isnull=True) | Q(end_date__gte=time_frame_start)).exists()
+
+    def in_numerator(self) -> bool:
+        """
+        Patients without active conditions within the list with ICD 10 I42.* I47.*, I48.*, I49.*.
+        """
+        # TODO: Settable time frame is not currently supported — where would it come from?
+        time_frame_provided = False
+        if not time_frame_provided:
+            # TODO: Should these datetimes be based on local time or UTC time?
+            time_frame_end = arrow.get()
+            time_frame_start = time_frame_end.shift(years=-1)
+        else:
+            raise RuntimeError("Time frame is not settable")
+
+        # TODO: Is the status field used at all for conditions, or is filtering done just with start/end dates?
+        qs = Condition.objects.filter(patient__id=self.patient_id, onset_date__lte=time_frame_end)
+
+        # TODO: still_active=False is not currently supported — where would the value come from?
+        still_active = True
+        if still_active:
+            return qs.filter(
+                Q(resolution_date__isnull=True) | Q(resolution__gte=time_frame_end)
+            ).exists()
+        else:
+            return qs.filter(
+                Q(resolution_date__isnull=True) | Q(resolution_date__gte=time_frame_start)
+            ).exists()
+
+    def compute(self) -> list[Effect]:
+        """Return a ProtocolCard effect if the patient is in both the denominator and numerator."""
+        if self.in_denominator():
+            if self.in_numerator():
+                first_name = Patient.objects.values_list("first_name").get(id=self.patient_id)[0]
+
+                title = "Consider updating the Conditions List to include Dysrhythmia related problem as clinically appropriate."
+                narrative = f"{first_name} has an active medication on the Medication List commonly used for Dysrhythmia. There is no associated condition on the Conditions List."
+
+                card = ProtocolCard(
+                    patient_id=self.patient_id,
+                    key="HCC004v1_RECOMMEND_DIAGNOSE_DYSRHYTHMIA",
+                    title=title,
+                    narrative=narrative,
+                    recommendations=[
+                        Recommendation(
+                            title=title,
+                            button="Diagnose",
+                            # TODO: Is the HREF value needed?
```

Comment:
@mbiannaccone

Reply:
href is not needed! its only used if there is no command and you want the button to link out to another site